File: src/test/java/nl/mpcjanssen/simpletask/util/UtilTest.java
package nl.mpcjanssen.simpletask.util;

import junit.framework.TestCase;

import nl.mpcjanssen.simpletask.task.Task;
import nl.mpcjanssen.simpletask.task.TodoListItem;

import java.util.ArrayList;

/**
 * Created with IntelliJ IDEA.
 * User: Mark Janssen
 * Date: 21-7-13
 * Time: 12:28
 */

public class UtilTest extends TestCase {
    public void testAddHeaderLines () {
        ArrayList<Task> tasks = new ArrayList<>();
        tasks.add(new Task("@Home h:1"));
        tasks.add(new Task("@Work h:1"));
        tasks.add(new Task("@College h:1"));
        ArrayList<TodoListItem> items = new ArrayList<>();
        for (Task t : tasks) {
            items.add(new TodoListItem(0,t,false));
        }
        assertEquals(0,Util.addHeaderLines(items, "by_context", "none", false, false).size());

    }
}

File: src/test/java/nl/mpcjanssen/simpletask/util/UtilTest.kt
package nl.mpcjanssen.simpletask.util

import junit.framework.TestCase
import nl.mpcjanssen.simpletask.dao.gentodo.TodoItem

import nl.mpcjanssen.simpletask.task.Task
import java.util.*

class UtilTest : TestCase() {
    fun testAddHeaderLines() {
        val tasks = ArrayList<Task>()
        tasks.add(Task("@Home h:1"))
        tasks.add(Task("@Work h:1"))
        tasks.add(Task("@College h:1"))
        val items = ArrayList<TodoItem>()
        for (t in tasks) {
            items.add(TodoItem(0, t, false))
        }
        assertEquals(6, addHeaderLines(items, ArrayList(listOf<String>("by_context")) , "none", false, null).size)

    }

    fun testIntersectUnion() {
        // Don't crash on an empty lists
        assertEquals(HashSet<String>(), ArrayList<HashSet<String>>().intersection())
        assertEquals(HashSet<String>(), ArrayList<HashSet<String>>().union())
    }
}
